intro/blockchain-intro-solution.js

```diff
@@ -0,0 +1,74 @@
+const SHA256 = require('crypto-js/sha256');
+
+// Block class
+function Block(index, timestamp, data, previousHash) {
+    this.index = index;
+    this.timestamp = timestamp;
+    this.data = data;
+    this.previousHash = previousHash;
+    this._nonce = 0;
+    this.hash = '';
+    
+    // Calculate a hash based on data supplied to block
+    this.calculateHash = function () {
+        return SHA256("" + this.index + this.previousHash + this.timestamp + JSON.stringify(this.data) + this._nonce).toString();
```

Comment:
Unfortunately, if you just `JSON.stringify` the data, it will be non-deterministic, and you won't get the same hash every time. You can pass in a sorted array of keys as the second parameter to sort the final JSON string. It must include _every_ key, including those of nested objects, or those keys will be omitted from the string.

We could do this manually in our solution code, or we could import this library instead of using `JSON.stringify`: https://github.com/substack/json-stable-stringify